src/test/kotlin/com/team4099/units/ChargeTest.kt

```diff
@@ -0,0 +1,189 @@
+import com.team4099.lib.units.derived.inZeptocoulombs
+import com.team4099.lib.units.derived.inZetacoulombs
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+
+/* Unit tests for Charge units */
+class ChargeTest {
+  private val kEpsilon = 1E-9
+  // Hacky way to do this. Consider removing support in the future for nano and below due to
+  // floating point imprecision
+  private val kDelta = 1E-3
+  private val kSigma = 1E6
+  private val kAlpha = 1E9
+  private val kBeta = 1E12
+
+  @Test
+  fun testCoulombsToYottacoulombs() {
+    val chargeCoulombs = 4099.coulombs
+    Assertions.assertEquals(chargeCoulombs.inYottacoulombs, 4.099E-21, kEpsilon)
+  }
+
+  @Test
+  fun testCoulombsToZetacoulombs() {
+    val chargeCoulombs = 4099.coulombs
+    Assertions.assertEquals(chargeCoulombs.inZetacoulombs, 4.099E-18, kEpsilon)
+  }
+
+  @Test
+  fun testCoulombsToExacoulombs() {
+    val chargeCoulombs = 4099.coulombs
+    Assertions.assertEquals(chargeCoulombs.inExacoulombs, 4.099E-15, kEpsilon)
+  }
+
+  @Test
+  fun testCoulombsToPetacoulombs() {
+    val chargeCoulombs = 4099.coulombs
+    Assertions.assertEquals(chargeCoulombs.inPetacoulombs, 4.099E-12, kEpsilon)
+  }
+
+  @Test
+  fun testCoulombsToTeracoulombs() {
+    val chargeCoulombs = 4099.coulombs
+    Assertions.assertEquals(chargeCoulombs.inTeracoulombs, 4.099E-9, kEpsilon)
+  }
+
+  @Test
+  fun testCoulombsToGigacoulombs() {
+    val chargeCoulombs = 4099.coulombs
+    Assertions.assertEquals(chargeCoulombs.inGigacoulombs, 4.099E-6, kEpsilon)
+  }
+
+  @Test
+  fun testCoulombsToMegacoulombs() {
+    val chargeCoulombs = 4099.coulombs
+    Assertions.assertEquals(chargeCoulombs.inMegacoulombs, 4.099E-3, kEpsilon)
+  }
+
+  @Test
+  fun testCoulombsToKilocoulombs() {
+    val chargeCoulombs = 4099.coulombs
+    Assertions.assertEquals(chargeCoulombs.inKilocoulombs, 4.099E0, kEpsilon)
+  }
+
+  @Test
+  fun testCoulombsToHectocoulombs() {
+    val chargeCoulombs = 4099.coulombs
+    Assertions.assertEquals(chargeCoulombs.inHectocoulombs, 4.099E1, kEpsilon)
+  }
+
+  @Test
+  fun testCoulombsToDecacoulombs() {
+    val chargeCoulombs = 4099.coulombs
+    Assertions.assertEquals(chargeCoulombs.inDecacoulombs, 4.099E2, kEpsilon)
+  }
+
+  @Test
+  fun testCoulombsToDecicoulombs() {
+    val chargeCoulombs = 4099.coulombs
+    Assertions.assertEquals(chargeCoulombs.inDecicoulombs, 4.099E4, kEpsilon)
+  }
+
+  @Test
+  fun testCoulombstoCenticoulombs() {
+    val chargeCoulombs = 4099.coulombs
+    Assertions.assertEquals(chargeCoulombs.inCenticoulombs, 4.099E5, kEpsilon)
+  }
+
+  @Test
+  fun testCoulombstoMillicoulombs() {
+    val chargeCoulombs = 4099.coulombs
+    Assertions.assertEquals(chargeCoulombs.inMillicoulombs, 4.099E6, kEpsilon)
+  }
+
+  @Test
+  fun testCoulombstoMicrocoulombs() {
+    val chargeCoulombs = 4099.coulombs
+    Assertions.assertEquals(chargeCoulombs.inMicrocoulombs, 4.099E9, kEpsilon)
+  }
+
+  // Unstable due to floating point errors
```

Comment:
This is confusing to me -- the way floats/doubles are represented in the machine is basically binary scientific notation, so it doesn't make sense to me that as the magnitude gets smaller the precision drops (I would expect precision to drop when you have too many significant figures)

Reply:
this problem exists because to convert to say `Nanocoulombs` we divide by `1.nano` and so instead of 1 / 1E-9 -> 1E9, it turns into 9.9999999E8. if we were to just multiply by 1E9 then there wouldn't be a loss of precision tho

Reply:
i'm not really sure how to solve this problem besides just storing the inverses of all the magnitudes